vector_store.py

```
@@ -8,10 +8,22 @@
 
 logger = logging.getLogger(__name__)
 
+
+def _chroma_call(fn, *args, timeout: float = 5.0, **kwargs) -> Any:
+    """Run a ChromaDB call with a timeout. Raises TimeoutError if it hangs."""
+    with concurrent.futures.ThreadPoolExecutor(max_workers=1) as ex:
+        fut = ex.submit(fn, *args, **kwargs)
+        try:
+            return fut.result(timeout=timeout)
+        except concurrent.futures.TimeoutError:
+            raise TimeoutError(f"ChromaDB operation timed out after {timeout}s")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)